app/src/main/java/com/mashup/lastgarden/ui/onboarding/OnBoardingActivity.kt

```diff
@@ -2,13 +2,51 @@ package com.mashup.lastgarden.ui.onboarding
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
-import com.mashup.lastgarden.R
+import androidx.viewpager2.widget.ViewPager2
+import com.google.android.material.tabs.TabLayoutMediator
+import com.mashup.lastgarden.databinding.ActivityOnBoardingBinding
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class OnBoardingActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityOnBoardingBinding
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_on_boarding)
+        binding = ActivityOnBoardingBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        setupViewPager()
+
+        binding.vpOnboarding.registerOnPageChangeCallback(object :
+            ViewPager2.OnPageChangeCallback() {
+
+            override fun onPageSelected(position: Int) {
+                super.onPageSelected(position)
+                when (binding.vpOnboarding.currentItem) {
+                    3 -> binding.btnNext.text = "시작하기"
+                    else -> binding.btnNext.text = "다음"
+
+                }
+            }
+        })
+    }
+
+    private fun setupViewPager() {
+        binding.vpOnboarding.adapter = TutorialViewPagerAdapter(this)
+        TabLayoutMediator(binding.tabLayout, binding.vpOnboarding) { tab, _ ->
+        }.attach()
+
+        binding.btnNext.setOnClickListener {
+            binding.vpOnboarding.currentItem++
+        }
+    }
+
+    override fun onBackPressed() {
+        if (binding.vpOnboarding.currentItem == 0) {
```

Comment:
```suggestion
        if (binding.vpOnboarding.currentItem <= 0) {
```
이렇게 작성하면 아래가 > 0 가 되기 때문에 버그 방어 코드가 될 것 같아요. 
이건 그냥 권유입니다!